src/cautiousrobot/utils.py

```diff
@@ -44,3 +46,37 @@ def process_csv(csv_path, expected_cols):
         raise Exception(f"The CSV at {csv_path} is missing column(s): {missing_cols}, defined as {[expected_cols[col] for col in missing_cols]}.")
     
     return df
+
+def downsample_and_save_image(image_dir_path, image_name, downsample_dir_path, downsample_size, log_errors, image_index, url, error_log_filepath):
+    """
+    Downsample an image and save it to the specified directory.
+
+    Parameters:
+    - image_dir_path (str): The path to the directory containing the original image.
+    - image_name (str): The name of the image to be downsampled.
+    - downsample_dir_path (str): The path to the directory where the downsampled image will be saved.
+    - downsample_size (int): The new size (both width and height) for the downsampled image.
+    - log_errors (list): A list to store errors encountered during the downsampling process.
```

Comment:
```suggestion
    - log_errors (dict): A dictionary to store errors encountered during the downsampling process.
```
This is the input and output of `log_response` (`log_data`).